rholang/src/main/scala/coop/rchain/rholang/interpreter/normalize.scala

```diff
@@ -234,7 +234,7 @@ object ProcNormalizeMatcher {
 
     def normalizeIfElse(valueProc: Proc,
                         trueBodyProc: Proc,
-                        falseBodyProc: Proc): M[ProcVisitOutputs] =
+                        falseBodyProc: Proc, input: ProcVisitInputs): M[ProcVisitOutputs] =
       for {
         targetResult <- normalizeMatch[M](valueProc, input)
```

Comment:
Wouldn't it be cleaner just to adjust the value here rather than passing input?

Reply:
Functions in FP should depend only on their input values.